src/tools/rustfmt/src/items.rs

```diff
@@ -106,11 +106,12 @@ impl Rewrite for ast::Local {
 
         result.push_str(&infix);
 
-        if let Some(ref ex) = self.init {
+        if let Some((init, _els)) = self.kind.init_else_opt() {
             // 1 = trailing semicolon;
             let nested_shape = shape.sub_width(1)?;
 
-            result = rewrite_assign_rhs(context, result, &**ex, nested_shape)?;
+            result = rewrite_assign_rhs(context, result, init, nested_shape)?;
+            // todo else
```

Comment:
This looks like rustfmt would butcher the let-else statements pretty badly. The typical first pass for new syntax with rustfmt often just involves spitting back out the unformatted/unprocessed contents of the original span, and I'd much prefer doing that here.

IIRC the span for locals doesn't include attributes so I'd suggest extending the condition above that checks for the skip attribute to also exit early if the local kind is InitElse. That should result in rustfmt emitting the statement and any attrs.

Don't feel obligated to do this, but if it helps you could add a snippet somewhere under `rustfmt/tests/target` either in a new file or an existing file that seems relevant, just to make sure that rustfmt emit invalid code that won't compile.

Reply:
Sounds good. I will replace the todo with the following and add a quick test, pending other comments.

```rust
if let Some(els) = els {
    result.push_str(" else ");
    result.push_str(context.snippet(els.span));
}
```

Reply:
No, I don't want to go that route. The formatting rules for the new syntax haven't been defined yet, and we don't want to be doing partial rewriting against an undefined spec.

We need to leave the original formatting in place for now. We'll update things once the corresponding formatting rules are codified and we've accounted for comments as well.

Extending the early exit check up above that starts on line 51 to include an or check for the corresponding localkind is what I'd want for the rustfmt piece, and which would be compatible with the existing changes/not require anything additional

```rust
        if contains_skip(&self.attrs) {
            return None;
        }
```

Reply:
Sorry 🤦 I didn't read your first comment closely. I'm too excited and need to slow down.